src/Features/CSharp/Portable/Copilot/CSharpImplementNotImplementedExceptionDiagnosticAnalyzer.cs

```diff
@@ -54,4 +71,58 @@ private void AnalyzeThrow(OperationAnalysisContext context, INamedTypeSymbol not
                 throwOperation.Syntax.GetLocation()));
         }
     }
+
+    private void AnalyzeMethod(SyntaxNodeAnalysisContext context, INamedTypeSymbol notImplementedExceptionType, HashSet<SyntaxNode> reportedMembers)
+    {
+        if (context.Node is not MemberDeclarationSyntax memberDeclaration || reportedMembers.Contains(memberDeclaration))
+            return;
+
+        var nameToken = GetMemberNameToken(memberDeclaration);
+        if (nameToken == default)
+            return;
+
+        var semanticModel = context.SemanticModel;
+        var throwNodes = memberDeclaration.DescendantNodes()
+            .Where(n => n is ThrowStatementSyntax || n is ThrowExpressionSyntax);
+
+        foreach (var throwNode in throwNodes)
+        {
+            ExpressionSyntax? expression = null;
+            if (throwNode is ThrowStatementSyntax throwStatement)
+                expression = throwStatement.Expression;
+            else if (throwNode is ThrowExpressionSyntax throwExpression)
+                expression = throwExpression.Expression;
+
+            if (expression is ObjectCreationExpressionSyntax objectCreation)
+            {
+                var typeInfo = semanticModel.GetTypeInfo(objectCreation);
+                if (typeInfo.Type != null && notImplementedExceptionType.Equals(typeInfo.Type))
+                {
+                    reportedMembers.Add(memberDeclaration);
+                    context.ReportDiagnostic(Diagnostic.Create(
+                        Descriptor,
+                        nameToken.GetLocation()));
+
+                    // Report only once for each member
+                    break;
+                }
+            }
+        }
+    }
+
+    private static SyntaxToken GetMemberNameToken(MemberDeclarationSyntax memberDeclaration)
```

Comment:
laternatively, get the ISymbol for the MemberDecl, and use .Location for it that intersects the span of hte MemberDecl.  it already has this logic.

Reply:
but I want the span where the name is declaration not only to check if it intersects

Reply:
>but I want the span where the name is declared

That's what ISymbol.Location is.  I'm just saying to find the intersecting one as a symbol can have multiple name locations :)